src/m3/deserializer.ts

```diff
@@ -0,0 +1,200 @@
+import {
+    Concept,
+    ConceptInterface,
+    Containment,
+    Datatype,
+    Feature,
+    FeaturesContainer,
+    Link,
+    M3Concept,
+    Metamodel,
+    MetamodelElement,
+    PrimitiveType,
+    Property,
+    Reference
+} from "./types.ts"
+import {SingleRef} from "../references.ts"
+import {Id, Node} from "../types.ts"
+import {SerializedNode} from "../serialization.ts"
+import {metaConcepts, metaFeatures} from "./self-definition.ts"
+
+
+export const deserializeMetamodel = (serializedNodes: SerializedNode[]): Metamodel => {
+
+    const metamodelSerNode = serializedNodes.find(({type}) => type === metaConcepts.metamodel.id)
+    if (metamodelSerNode === undefined) {
+        throw new Error(`could not deserialize: no instance of LIonCore's Metamodel concept found in serialization`)
+    }
+
+    // make a map id -> serialized node:
+    const serializedNodeById: { [id: Id]: SerializedNode } = {}
+    serializedNodes.forEach((node) => {
+        serializedNodeById[node.id] = node
+    })
+
+    const deserializedNodeById: { [id: Id]: Node } = {}
+
+    const constructMemoised = (serNode: SerializedNode, parent?: M3Concept): M3Concept => {
+        const node = construct(serNode, parent)
+        deserializedNodeById[node.id] = node
+        return node
+    }
+
+    const mapConstructMemoised = <T extends Node>(parent: M3Concept) =>
```

Comment:
The role of this element is not clear to me. A comment could help

Reply:
Agreed: pushing a commit that documents this code.